libs/execution_context/EBPF_NATIVE.H

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "ebpf_core_structs.h"
+#include "ebpf_maps.h"
+#include "bpf2c.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+    typedef struct _ebpf_native ebpf_native_t;
+
+    /**
+     * @brief Initialize the eBPF native code module.
+     *
+     * @retval EBPF_SUCCESS The operation was successful.
+     * @retval EBPF_NO_MEMORY Unable to allocate resources for this
+     *  operation.
+     */
+    ebpf_result_t
+    ebpf_native_initiate();
+
+    /**
+     * @brief Uninitialize the eBPF native code module.
+     *
+     */
+    void
+    ebpf_native_terminate();
+
+    /**
+     * @brief Load native module driver and bind to the native module.
+     *
+     * @param[in] service_name Name of the service for the native module.
+     * @param[in] service_name_length Length of service_name.
+     * @param[in] module_id Identifier of the native eBPF module to load.
+     * @param[out] count_of_maps Count of maps in the native module.
+     * @param[out] count_of_programs Count of programs in the native module.
+     *
+     * @retval EBPF_SUCCESS The operation was successful.
+     * @retval EBPF_NO_MEMORY Unable to allocate resources for this
+     *  operation.
+     * @retval EBPF_OBJECT_NOT_FOUND Native module for that module ID not found.
+     * @retval EBPF_OBJECT_ALREADY_EXISTS Native module for this module ID is already
+     *  initialized.
+     */
+    ebpf_result_t
+    ebpf_native_load(
+        _In_ const wchar_t* service_name,
```

Comment:
if this is not null terminated then:
```suggestion
        _In_reads_(service_name_length) const wchar_t* service_name,
```

if it is null terminated then you don't need the service_name_length param